src/main/java/dk/cngroup/lentils/service/TeamService.java

```diff
@@ -0,0 +1,78 @@
+package dk.cngroup.lentils.service;
+
+import dk.cngroup.lentils.entity.Team;
+import dk.cngroup.lentils.repository.TeamRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.util.Optionals;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+
+public class TeamService {
+
+    TeamRepository repository;
+
+    @Autowired
+    public TeamService(TeamRepository repository) {
+        this.repository = repository;
+    }
+
+    public Team add(Team team) {
+        team.setPin(getUniquePin());
+        return repository.save(team);
+    }
+
+    public Optional<Team> get(long id) {
+        return repository.findById(id);
+    }
+    public void delete(Team team) {
+        repository.delete(team);
+    }
+
+    public void deleteAll() {
+        repository.deleteAll();
+    }
+
+    public List<Team> getAll() {
+        return repository.findAll();
+    }
+
+    private String getUniquePin() {
+        int pinLen = 4;
+        String pin = getPIN(pinLen);
+
+        while (!PINisUnique(pin)) {
+            pin = getPIN(pinLen);
+        }
+        return pin;
+    }
+
+    private boolean PINisUnique(String pin) {
```

Comment:
Ta entita by pak nemela obsahovat tuhle dependency na repository

Reply:
To jsem si nevšiml. To je dobrý point, i s tou factory.